### ISSUE TITLE
Regression in patch for #181

### ISSUE THREAD

**Issue by a1lu** [eclipse/paho.mqtt-sn.embedded-c#183]
Hi, 
I noticed a regression in 1b08591fad433b47bc0b87c4d9455a19187e3c18.

I try to connect a client to the gateway. Without the patch:

```
20200220 092239.567   CONNECT           <---  clientId                            0E 04 04 01 00 3C 63 6C 69 65 6E 74 49 64
20200220 092239.567   CONNECT           ===>  clientId                            10 14 00 04 4D 51 54 54 04 02 00 3C 00 08 63 6C 69 65 6E 74 49 64
20200220 092239.689   CONNACK           <===  clientId                            20 02 00 00
20200220 092239.689   CONNACK           --->  clientId                            03 05 00

20200220 092239.689   SUBSCRIBE   0001  <---  clientId                            09 12 00 00 01 74 65 73 74
20200220 092239.690   SUBSCRIBE   0001  ===>  clientId                            82 09 00 01 00 04 74 65 73 74 00
20200220 092239.690   SUBACK      0001  <===  clientId                            90 03 00 01 00
20200220 092239.690   SUBACK      0001  --->  clientId                            08 13 00 00 01 00 01 00
```

With the patch:
```
20200220 092511.198   CONNECT           <---  clientId                            0E 04 04 01 00 3C 63 6C 69 65 6E 74 49 64
20200220 092511.198   CONNECT           ===>  clientId                            10 14 00 04 4D 51 54 54 04 02 00 3C 00 08 63 6C 69 65 6E 74 49 64
20200220 092511.208   CONNACK           <===  clientId                            20 02 00 00
20200220 092511.208   CONNACK           --->  clientId                            03 05 00

20200220 092511.209   SUBSCRIBE   0001  <---  Unknown Client !                    09 12 00 00 01 74 65 73 74
```

Regards

**Comment by ty4tw** [comment]
Hi,

I could not reproduce the error.
![image](https://user-images.githubusercontent.com/7830788/74932421-1de08d00-5425-11ea-82f6-6ec502cdae9d.png)

**Comment by a1lu** [comment]
Do you have an entry for the client in clients.conf?

**Comment by ty4tw** [comment]
ClientList file name was not shown on the log.
I fixed this bug and push it.

But the bugfix of #181 is working. 
I guess your client.conf was over written by  "make install".
Please check it with latest develop branch.
Thank you for your cooperation.

**Comment by ty4tw** [comment]
The bug occurred in case of QoS-1=YES in gateway.conf.

**Comment by a1lu** [comment]
Hi,
I can confirm that this is fixed in 1b6e6faab5fc6f3162cf1d07b2d658cc1af93e9e

thanks.
